extensions/plugins/system/redcore/redcore.php

```diff
@@ -90,6 +90,11 @@ public function onAfterInitialise()
 						$data = RApi::getPostedData();
 						$dataGet = $input->get->getArray();
 
+						if (!empty($data['language']) && empty($dataGet['language']))
```

Comment:
Hmm... I answered this question in the chat... just gonna c/p here :)

Kristijan [07:40] 
There is a simple solution to your problem as I just read about it... The issue is that translations are turned on / off by default even before API is set to running, at the absolute start of the redCORE plugin. So if the language parameter is not in the GET parameters it will not be turned on. In SOAP we use language as a parameter for the functions (for SOAP server handle function). So what we need to do is to see if language parameter in SOAP read operation is set, if it is then we manually put it into a JFactory::getApplication()->input which should cascade all translation triggers. (That part with enabling it for admin is correct when using API, but not just soap, for HAL too...)